contracts/koh/test/ProposalsTest.js

```diff
@@ -0,0 +1,271 @@
+let catchRevert            = require("./exceptions.js").catchRevert;
+let catchOutOfGas          = require("./exceptions.js").catchOutOfGas;
+let catchInvalidJump       = require("./exceptions.js").catchInvalidJump;
+let catchInvalidOpcode     = require("./exceptions.js").catchInvalidOpcode;
+let catchStackOverflow     = require("./exceptions.js").catchStackOverflow;
+let catchStackUnderflow    = require("./exceptions.js").catchStackUnderflow;
+let catchStaticStateChange = require("./exceptions.js").catchStaticStateChange;
+
+
+describe('TestKingAutomatonProposals1', async() => {
+  const KingAutomaton = artifacts.require("KingAutomaton");
+
+  beforeEach(async() => {
+    accounts = await web3.eth.getAccounts();
+    account = accounts[0];
+    koh = await KingAutomaton.new(4, 16, "0x010000", "406080000");
+    id = await koh.createProposal.call(account, "", "", "0x");
+    await koh.createProposal(account, "", "", "0x");
+    proposal_start_period = 90;
+    contest_period = 90;
+  });
+
+  it("correct proposal creation", async() => {
+    assert.equal(id, 2, "Incorrect ID!");
+    assert.exists(account, "Account doesn't exist!");
+    assert.exists(koh.address, "Contract wasn't deployed!");
+    let ballot = await koh.ballot_boxes(id);
+    let ballot_state = ballot.state;
+    assert.equal(ballot_state, 1, "Ballot box state is not PrepayingGas!");
+    let proposal = await koh.proposals(id)
+    let proposal_state = proposal.state;
+    assert.equal(proposal_state, 1, "Proposal state is not Started!");
+  });
+
+  it("cast vote invalid slot owner", async() => {
+    await koh.payForGas(id, 3);
+    await catchRevert(koh.castVote(id, 0, 1), "Invalid slot owner!");
+  });
+
+  it("cast vote invalid ballot box id", async() => {
+    await koh.setOwnerAllSlots();
+    await catchRevert(koh.castVote(200, 0, 1), "Invalid ballot box ID!");
+  });
+
+  it("cast vote before gas is paid", async() => {
+    await koh.setOwnerAllSlots();
+    await catchRevert(koh.castVote(id, 0, 1), "Ballot is not active!");
+  });
+
+  it("pay gas for more slots", async() => {
+    await catchRevert(koh.payForGas(id, 20), "Too many slots!");
+  });
+
+  it("vote change", async() => {
+    await koh.setOwnerAllSlots();
+    await koh.payForGas(id, 3);
+    // Test initial vote values
+    let vote = await koh.getVote(id, 0);
+    let pos_vote_count = await koh.getVoteCount(id, 1);
+    let neg_vote_count = await koh.getVoteCount(id, 2);
+    assert.equal(vote, 0, "Incorrect initial vote!");
+    assert.equal(pos_vote_count, 0, "Incorrect initial positive vote count!");
+    assert.equal(neg_vote_count, 0, "Incorrect initial negative vote count!");
+    // Cast positive vote
+    await koh.castVote(id, 0, 1);
```

Comment:
Make VOTE_YES and VOTE_NO constants you can use for readability throughout the test.
`await koh.castVote(id, 0, 1);` would become `await koh.castVote(id, 0, VOTE_YES);`